bundles/org.openhab.binding.liquidcheck/src/main/java/org/openhab/binding/liquidcheck/internal/httpclient/LiquidCheckHttpClient.java

```diff
@@ -0,0 +1,119 @@
+/**
+ * Copyright (c) 2010-2022 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.liquidcheck.internal.httpclient;
+
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeoutException;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jetty.client.HttpClient;
+import org.eclipse.jetty.client.api.ContentResponse;
+import org.eclipse.jetty.client.api.Request;
+import org.eclipse.jetty.client.util.StringContentProvider;
+import org.eclipse.jetty.http.HttpHeader;
+import org.eclipse.jetty.http.HttpMethod;
+import org.openhab.binding.liquidcheck.internal.LiquidCheckConfiguration;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link LiquidCheckHttpClient} sets up the jetty client for the connection to the device.
+ *
+ * @author Marcel Goerentz - Initial contribution
+ */
+@NonNullByDefault
+public class LiquidCheckHttpClient {
+
+    private final Logger logger = LoggerFactory.getLogger(LiquidCheckHttpClient.class);
+    private final HttpClient client;
+    private final LiquidCheckConfiguration config;
+
+    public boolean isClosed = false;
+
+    /**
+     * The Constructor of the LiquidCheckHttpClient class will set up a jetty client
+     * 
+     * @param config
+     */
+    public LiquidCheckHttpClient(LiquidCheckConfiguration config) {
+        this.config = config;
+        client = new HttpClient();
```

Comment:
Do you really need to build your own HTTP client ?
And not use the common HTTP client available for all bindings ?
The "properties" you are setting for your HTTP client can be set in each request, I believe. I will check that.

Reply:
I believe you could use the common HTTP client.
When building your request object, you could specify something like  `request.timeout(config.connectionTimeout, TimeUnit.SECONDS).followRedirects(true)` to activate URL redirection and set a specific timeout for your request.

When you use the common HTTP client, you must not start and stop it in your binding.
You can get the common HTTP client in your thing handler factory like that:
```
this.httpClient = httpClientFactory.getCommonHttpClient();
```
You can find examples in many bindings.

Reply:
Here is an example: https://github.com/openhab/openhab-addons/blob/main/bundles/org.openhab.binding.remoteopenhab/src/main/java/org/openhab/binding/remoteopenhab/internal/RemoteopenhabHandlerFactory.java#L94